Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLParameterScan.py

```diff
@@ -0,0 +1,277 @@
+
+    def PyExec(self):
+
+        self.setUp()
+
+        _, load_ws_name = needs_loading(self.sample, "Load")
+        Load(Filename=self.sample, OutputWorkspace=load_ws_name)
+        ConjoinXRuns(InputWorkspaces=load_ws_name, OutputWorkspace="__joined", SampleLogAsXAxis=self.observable)
+        mtd[load_ws_name].delete()
+
+        sort_x_axis_output = 'sorted'
+        SortXAxis(InputWorkspace="__joined", OutputWorkspace="__" + sort_x_axis_output)
+        mtd["__" + sort_x_axis_output].getAxis(0).setUnit("label").setLabel('Omega', 'degrees')
+
+        load_sensitivity, sens_input = needs_loading(self.sensitivity, 'Sensitivity')
+        self.progress.report('Loading sensitivity')
+        if load_sensitivity:
+            LoadNexusProcessed(Filename=self.sensitivity, OutputWorkspace=sens_input)
+
+        load_mask, mask_input = needs_loading(self.mask, "Mask")
+        self.progress.report('Loading mask')
+        if load_mask:
+            LoadNexusProcessed(Filename=self.mask, OutputWorkspace=mask_input)
+
+        load_default_mask, default_mask_input = needs_loading(self.default_mask, "DefaultMask")
+        self.progress.report('Loading default mask')
+        if load_default_mask:
+            LoadNexusProcessed(Filename=self.default_mask, OutputWorkspace=default_mask_input)
+
+        process_absorber, absorber_name = needs_processing(self.absorber, 'Absorber')
+        self.progress.report('Processing absorber')
+        if process_absorber:
+            SANSILLReduction(Run=self.absorber,
+                             ProcessAs='Absorber',
+                             NormaliseBy=self.normalise,
+                             OutputWorkspace=absorber_name)
+
+        process_container, container_name = needs_processing(self.container, 'Container')
+
+        self.progress.report('Processing container')
+        if process_container:
+            SANSILLReduction(Run=self.container,
+                             ProcessAs='Container',
+                             OutputWorkspace=container_name,
+                             AbsorberInputWorkspace=absorber_name,
+                             CacheSolidAngle=True,
+                             NormaliseBy=self.normalise)
+        # even if the samples are provided as arguments, the SANSILLReduction won't use them and rather take the already
+        # processed ws
+        SANSILLReduction(Run=self.sample,
+                         AbsorberInputWorkspace=absorber_name,
+                         ContainerInputWorkspace=container_name,
+                         SampleThickness=self.thickness,
+                         SensitivityInputWorkspace=sens_input,
+                         SensitivityOutputWorkspace=self.output_sens,
+                         MaskedInputWorkspace=mask_input,
+                         DefaultMaskedInputWorkspace=default_mask_input,
+                         NormaliseBy=self.normalise,
+                         OutputWorkspace=sort_x_axis_output)
```

Comment:
I think it would be useful to output also this workspace, right before detector grouping.